main.go

```diff
@@ -19,9 +19,56 @@ limitations under the License.
 package main
 
 import (
-	"fmt"
+	"flag"
+	"os"
+
+	"sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/controllers"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/api/v1alpha1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	// +kubebuilder:scaffold:imports
 )
 
+var (
+	myscheme = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+func init() {
+	_ = scheme.AddToScheme(myscheme)
+	_ = v1alpha1.AddToScheme(myscheme)
```

Comment:
is this something that kubebuilder requires?
would it matter if these calls are under `var (` or `main()` with the idea to not have an `init()` function in this file?

Reply:
this was mostly a straight port from the tube builder book. Doesn't matter where it is to be honest. I think this is for the `// +kubebuilder:scaffold:scheme` thing, which I don't actually know what that is...Let me look into that and see if we need this to be its own section or what.

Reply:
Ah yes, here it is: https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/scaffold/v2/main.go#L66

This section doesn't have to be in init() but if we end up generating more types the scaffold will try and put stuff here. Might as well keep it how they intended it to be imo

Reply:
ok, understood. having `init()` in main.go in general seems redundant, but if the tooling uses it, then it's fine.